addons/account/models/account_move.py

```diff
@@ -1900,9 +1924,13 @@ def write(self, vals):
         return res
 
     @api.ondelete(at_uninstall=False)
-    def _unlink_except_posted_before(self):
-        if not self._context.get('force_delete') and any(move.posted_before for move in self):
-            raise UserError(_("You cannot delete an entry which has been posted once."))
+    def _unlink_forbid_parts_of_chain(self):
+        """ Moves with a sequence number can only be deleted if they are the last element of a chain of sequence.
+        If they are not, deleting them would create a gap. If the user really wants to do this, he still can
+        explicitly empty the 'name' field of the move; but we discourage that practice.
+        """
+        if not self._context.get('force_delete') and any(move.name != '/' and not move._is_last_from_seq_chain() for move in self):
+            raise UserError(_("You cannot delete this entry, as it has already consumed a sequence number and is not the last one in the chain. Probably you should revert it instead."))
```

Comment:
Maybe, but I'm not sure. For moves, it's really important, but is it for bank statements ? Or any other object inheriting from the mixin in the future ? Plus, here, it's interesting to report that because the user probably should have made a reverse move; and there probably won't be any equivalent for that on other objects ; so, should we make deletion so painful for them as well ?

Reply:
The mixin is implemented like a no gap sequence, so it definitely makes sense to add it; if it evolves later to be possible to have gaps, then it will be easy to adapt the constraint